h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -304,6 +304,24 @@ protected GLMModel deepClone(Key<GLMModel> result) {
     public boolean _generate_scoring_history = false; // if true, will generate scoring history but will slow algo down
     
     public void validate(GLM glm) {
+      if (_remove_collinear_columns) {
+        if (!(Solver.IRLSM.equals(_solver) || Solver.AUTO.equals(_solver)))
+        glm.warn("remove_collinear_columns", "remove_collinear_columns only works when IRLSM (or " +
+                "AUTO which will default to IRLSM) is chosen as the solver.  Otherwise, remove_collinear_columns is " +
```

Comment:
```suggestion
                "AUTO which will default to IRLSM when the number of predictors is small) is chosen as the solver.  Otherwise, remove_collinear_columns is " +
```

Reply:
@wendycwong IRLSM is only the default sometimes, so maybe it's better to clarify that.  The docs doesn't say what the cutoff for the number of predictors is to switch from IRLSM to L-BFGS, so I just kept it generic, "number of predictors is small".  What do you think?

Reply:
@ledell I actually set IRLSM as the default solver if remove_collinear_columns = true regardless of anything else.  The logic given to me by michalk is this:  if the user doesn't choose the solver, that means s/he does not care and we will choose.  Hence, if remove_collinear_columns = true, it will dictate the solver type to IRLSM regardless of any other parameters settings.  If remove_collinear_columns = false, it will choose the solver according to the rules that are already there.  What do you think of how the solver is chosen?

Reply:
@ledell This is the new warning message now:

RuntimeWarning: remove_collinear_columns only works when IRLSM (or AUTO which will default to IRLSM when remove_collinear_columns=true) is chosen as the solver.  Otherwise, remove_collinear_columns is not enabled.